examples/step-95/step-95.cc

```diff
@@ -0,0 +1,1711 @@
+      const auto normal_gradient_term = -value;
+
+      evaluator_m.submit_value(value_term, q);
+      evaluator_m.submit_normal_derivative(normal_gradient_term, q);
+    }
+
+    // The implementation of the face-based ghost penalty term (up to degree 2 /
+    // normal hessians).
+    template <bool do_normal_hessians, bool do_values, typename Integrator>
+    inline void do_gp_face_term(
+      Integrator                            &evaluator_m,
+      Integrator                            &evaluator_p,
+      const typename Integrator::NumberType &masked_factor_value,
+      const typename Integrator::NumberType &masked_factor_gradient,
+      const typename Integrator::NumberType &masked_factor_hessian,
+      const unsigned int                     q) const
+    {
+      if (do_values)
+        {
+          const auto value_m = evaluator_m.get_value(q);
+          const auto value_p = evaluator_p.get_value(q);
+
+          const auto jump_value = value_m - value_p;
+
+          const auto value_term = masked_factor_value * jump_value;
+
+          evaluator_m.submit_value(value_term, q);
+          evaluator_p.submit_value(-value_term, q);
+        }
+
+      {
+        const auto normal_gradient_m = evaluator_m.get_normal_derivative(q);
+        const auto normal_gradient_p = evaluator_p.get_normal_derivative(q);
+
+        const auto jump_normal_gradient = normal_gradient_m - normal_gradient_p;
+
+        const auto gradient_term =
+          masked_factor_gradient * jump_normal_gradient;
+
+        evaluator_m.submit_normal_derivative(gradient_term, q);
+        evaluator_p.submit_normal_derivative(-gradient_term, q);
+      }
+
+      if (do_normal_hessians)
+        {
+          const auto normal_hessian_m = evaluator_m.get_normal_hessian(q);
+          const auto normal_hessian_p = evaluator_p.get_normal_hessian(q);
+
+          const auto jump_normal_hessian = normal_hessian_m - normal_hessian_p;
+
+          const auto hessian_term = masked_factor_hessian * jump_normal_hessian;
+
+          evaluator_m.submit_normal_hessian(hessian_term, q);
+          evaluator_p.submit_normal_hessian(-hessian_term, q);
+        }
+    }
+
+    // The implementation of the right-hand-side term evaluating the rhs
+    // function (unfortunately we cannot evaluate a Function vectorized, so we
+    // have to reshuffle the quadrature point data).
+    template <typename Integrator>
+    inline void do_rhs_cell_term(Integrator          &evaluator,
+                                 const Function<dim> &rhs_function,
+                                 const unsigned int   q) const
+    {
+      const auto q_points = evaluator.quadrature_point(q);
+
+      VectorizedArrayType value = 0.;
+
+      for (unsigned int v = 0; v < n_lanes; ++v)
+        {
+          Point<dim> q_point;
+          for (unsigned int d = 0; d < dim; ++d)
+            q_point[d] = q_points[d][v];
+
+          value[v] = rhs_function.value(q_point);
```

Comment:
We have this or similar code in various places, e.g. in the step-67 tutorial https://github.com/dealii/dealii/blob/6b1fb51c0b1fb26bbceff0d5d0f292855a1e3b49/examples/step-67/step-67.cc#L631-L646 It could make sense to create a helper function at a general place for this in the library. (Not necessarily now, but we should do it I believe.)

Reply:
This is something that bothers me because it should be easily possible. Unfortunately, my effort in #16328 showed that the usage of `Function` is too involved throughout the library to make this work.

So, I agree, we should provide a helper function. But this should be a follow up that updates all relevant places.